mne/viz/_figure.py

```diff
@@ -2351,6 +2351,21 @@ def _psd_figure(inst, proj, picks, axes, area_mode, tmin, tmax, fmin, fmax,
     titles_list = list()
     scalings_list = list()
     psd_list = list()
+    # exclude channels
+    bad_ch_idx = [inst.info['ch_names'].index(ch) for ch in inst.info['bads']
+                  if ch in inst.info['ch_names']]
+    if len(exclude) > 0:
+        if isinstance(exclude, str) and exclude == 'bads':
+            exclude = bad_ch_idx
+        elif (isinstance(exclude, list) and
+              all(isinstance(ch, str) for ch in exclude)):
+            exclude = [inst.info['ch_names'].index(ch) for ch in exclude]
+        else:
+            raise ValueError(
+                'exclude has to be a list of channel names or "bads"')
+
+        picks = np.array([pick for pick in picks if pick not in exclude])
```

Comment:
We can use some internal functions to simplify this. We should probably also check that `include` are valid channel names
```suggestion
    inst.info._check_consistency()
    _validate_type(exclude, (list, tuple, str), 'exclude')
    if isinstance(exclude, str):
        _check_option('exclude', exclude, 'bads', extra='when str')
        exclude = info['bads']
    for ei, ex in enumerate(exclude):
        if not isinstance(ex, str) or ex not in info['ch_names']:
            raise ValueError(f'exclude[{ei}] ({repr(ex)}) '
                             'not found in info["ch_names"]')
    if len(exclude) > 0:
        picks = np.array([pick for pick in picks if
                          info['ch_names'][pick] not in exclude], int)
```

Reply:
Looks good to me!

As I simply copied the code from the erp.plot(), should I port the suggested change here, to that function as well?